includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -2276,4 +2338,161 @@ protected function guess_modal_role( DOMElement $modal ) {
 		// None of the roles we are looking for match any of the classes.
 		return Role::DIALOG;
 	}
+
+	/**
+	 * Evaluates the given XPath expression and give first Element if exist.
+	 *
+	 * @param string  $expression   The XPath expression to execute.
+	 * @param Element $context_node The optional context node can be specified for doing relative XPath queries.
+	 *                              By default, the queries are relative to the root element.
+	 *
+	 * @return Element|bool Return Element if exists otherwise False.
+	 */
+	protected function get_first_element( $expression, $context_node = null ) {
+
+		if ( empty( $expression ) ) {
+			return false;
+		}
+
+		$dom_node_list = $this->dom->xpath->query( $expression, $context_node );
+		if ( ! $dom_node_list instanceof DOMNodeList || 0 >= $dom_node_list->length ) {
+			return false;
+		}
+
+		/** @var Element $dom_node */
+		$dom_node = $dom_node_list->item( 0 );
+
+		return $dom_node;
```

Comment:
```suggestion
		if ( ! $dom_node_list instanceof DOMNodeList ) {
			return null;
		}

		return $dom_node_list->item( 0 );
```

Reply:
Then again, if `$dom_node_list` is not a `DOMNodeList`, we'd want to know about it. So maybe this should rather be just:

```suggestion
		return $dom_node_list->item( 0 );
```